### ISSUE TITLE
allow querying gpt multiple times simultaneously

### ISSUE THREAD

**Issue by smallstepman** [nhaouari/obsidian-textgenerator-plugin#237]
perhaps it would be possible to use multiple cursors? 
maybe that's helpful: my emacs's gptel plugin does this by storying bounds describing LoC where chatgpt generated the response
```
<!-- gptel--bounds: ((369 . 388) (404 . 406) (436 . 452) (933 . 1166) (3978 . 3979) (4200 . 4201) (4731 . 4749) (4776 . 4782) (5246 . 7694) (8065 . 11808) (11816 . 13659) (13667 . 16599) (16607 . 19030) (19038 . 21505) (21513 . 24571) (24579 . 27892) (27900 . 31052)) -->
```
here is how it looks like

https://github.com/nhaouari/obsidian-textgenerator-plugin/assets/21069150/e99107ee-e8a4-44b9-887a-e7b327782607

**Comment by haouarihk** [comment]
i see,
the main draw back of this is that, it might over complicate the code responsible for stopping the stream (controller).
every time it starts a generation, it overrides the global controller.
but that's just technical and can be solved.

another issue is how we are going to control where it will go,
as you know, we use the multi select cursor for context handling, we can't reuse it for multi generation.

i'm happy to hear more about it

**Comment by smallstepman** [comment]
> the main draw back of this is that, it might over complicate the code responsible for stopping the stream (controller).

You’re right. The way I “deal” with this problem is I’ll just allow it to finish, and I’ll submit another (corrected, or not) query in the meantime. Feels easier than trying to go into some menu and attempt to select the correct query execution to stop. The mental model I’ve adapted to cope with using this strategy is “oh well, there goes another $0.04, too bad”. I’ll manually delete the response once the bad query finishes. 

Sadly I know nothing about the internals of this plugin, or the API provided by Obsidian for plugins. I won’t be much of a help in figuring out if this is even possible, let alone how to implement it. 

I’m not certain what do you mean by “where it will go”

**Comment by haouarihk** [comment]
you can make a template that generates multiple time,
or generates in sequence and result from one goes to another one,

what i mean by where it will go is , how is it gonna be triggered, and how does it know where to put the generation.
because the multiple cursor select is already being used
